server/index.ts

```diff
@@ -355,74 +402,19 @@ if (contentProxy) {
     }
   );
   app.get("/*/docs/*", async (req, res) => {
-    let lookupURL = decodeURI(req.path);
-    let extraSuffix = "";
-    let isMetadata = false;
-    let isDocument = false;
-
-    if (req.path.endsWith("index.json")) {
-      // It's a bit special then.
-      // The URL like me something like
-      // /en-US/docs/HTML/Global_attributes/index.json
-      // and that won't be found in getRedirectUrl() since that doesn't
-      // index things with the '/index.json' suffix. So we need to
-      // temporarily remove it and remember to but it back when we're done.
-      isDocument = true;
-      extraSuffix = "/index.json";
-      lookupURL = lookupURL.replace(extraSuffix, "");
-    } else if (req.path.endsWith("metadata.json")) {
-      isMetadata = true;
-      extraSuffix = "/metadata.json";
-      lookupURL = lookupURL.replace(extraSuffix, "");
-    }
-
-    const isJSONRequest = extraSuffix.endsWith(".json");
-
-    let document;
-    try {
-      console.time(`buildDocumentFromURL(${lookupURL})`);
-      document = await buildDocumentFromURL(lookupURL);
-    } catch (error) {
-      console.error(`Error in buildDocumentFromURL(${lookupURL})`, error);
-      return res.status(500).send(error.toString());
-    } finally {
-      console.timeEnd(`buildDocumentFromURL(${lookupURL})`);
-    }
-
-    if (!document) {
-      const redirectURL = Redirect.resolve(lookupURL);
-      if (redirectURL !== lookupURL) {
-        return res.redirect(301, redirectURL + extraSuffix);
-      }
-      return res
-        .status(404)
-        .sendFile(path.join(STATIC_ROOT, "en-us", "_spas", "404.html"));
-    }
-
-    if (isDocument) {
-      res.json({ doc: document });
-    } else if (isMetadata) {
-      const docString = JSON.stringify({ doc: document });
-
-      const hash = crypto.createHash("sha256").update(docString).digest("hex");
-      const { body: _, toc: __, sidebarHTML: ___, ...builtMetadata } = document;
-      builtMetadata.hash = hash;
-
-      res.json(builtMetadata);
-    } else if (isJSONRequest) {
-      // TODO: what's this for?
-      res.json({ doc: document });
-    } else {
-      res.header("Content-Security-Policy", CSP_VALUE);
-      res.send(renderHTML({ doc: document, url: lookupURL }));
+    const url = req.path;
+    const doc = await buildDocumentFromURL(url);
+    if (!doc) {
+      return redirectOr404(url);
     }
+    res.header("Content-Security-Policy", CSP_VALUE);
+    return res.send(renderHTML(doc));
```

Comment:
I don't think `doc` has a `url` key, does it?

Reply:
Indeed, the old world.